src/irmin/tree_intf.ml

```diff
@@ -326,20 +326,19 @@ module type S = sig
   (** {1 Proofs} *)
 
   module Proof : sig
```

Comment:
Is that `Proof` api really necessary? Aren't `produce_proof` and `verify_proof` enough?

Reply:
It's useful in the tests so I'd like to keep them (from now)